test/S3.jl

```diff
@@ -95,431 +94,87 @@ end
         end
+    end
 
-        if "S3" in ONLINE
-            @testset "Online" begin
-                @info "Running ONLINE S3 tests"
-
-                # Create bucket for tests
-                test_run_id = string(uuid4())
-                s3_prefix = if TEST_BUCKET_AND_PREFIX === nothing
-                    bucket = string("awstools-s3-test-temp-", uuid4())
-                    @info "Creating S3 bucket $bucket"
-                    s3_create_bucket(bucket)
-                    "s3://$bucket/$test_run_id"
-                else
-                    "$TEST_BUCKET_AND_PREFIX/$test_run_id"
-                end
-
-                try
-                    @testset "Upload to S3" begin
-                        dest = S3Path("$s3_prefix/folder3/testfile")
-
-                        try
-                            mktemp() do src, stream
-                                write(stream, "Local file src")
-                                close(stream)
-
-                                @test !exists(dest)
-
-                                uploaded_file = upload(Path(src), dest)
-                                @test isa(uploaded_file, S3Path)
-
-                                @test exists(dest)
-                                @test isfile(dest)
-                                @test isdir(parent(dest))
-                                @test read(dest, String) == "Local file src"
-                            end
-                        finally
-                            rm(dest; recursive=true)
-                        end
-                    end
-
-                    @testset "Download from S3" begin
-                        src = S3Path("$s3_prefix/folder4/testfile")
-
-                        try
-                            s3_put(src.bucket, src.key, "Remote content")
-
-                            @testset "Download to a directory" begin
-                                mktempdir() do dest_dir
-                                    dest = Path(dest_dir)
-
-                                    dest_file = download(src, dest)
-                                    @test isa(dest_file, AbstractPath)
-
-                                    @test exists(Path(dest_file))
-                                    @test read(dest_file, String) == "Remote content"
-                                end
-                            end
-
-                            @testset "Download to a local file" begin
-                                mktemp() do dest_file, stream
-                                    dest = Path(dest_file)
-                                    close(stream)
-
-                                    dest_file = download(src, dest)
-                                    @test isa(dest_file, AbstractPath)
-
-                                    @test dest_file == dest
-                                    @test read(dest, String) == "Remote content"
-                                end
-                            end
+    if "S3" in ONLINE
```

Comment:
With the exception of the addition of a couple `@test_deprecated`s all the changes from here down are either deletions or whitespace changes(due to deletions).